### DISCUSSION TITLE
Clickable file folder icon inside of file directory

### DISCUSSION THREAD

**Discussion by divinepersuasionstudio** [directus/directus#18649] [Feature Requests]
### Summary

Clickable file folder icon inside of file directory with folders displayed first and files displayed second. 

![Screenshot 2023-05-18 at 10 24 39 AM](https://github.com/directus/directus/assets/123796138/7f2fa6d0-c0d7-49c8-b112-e90ec3660f60)

### Basic Example

I would like to be able to see both folders and files within a parent folder in the same window.

### Motivation

To make it similar to their normal windows/mac/dropbox experience. This will provide our clients and users with an easy and familiar experience.

### Detailed Design

double click to go into nested folders where folder icons are listed along with the files in the folder

### Must Have:
- Icon 
- Clickability

### Should Have:
- folders listed first and files listed second

### Could Have:
- custom sorting options

### Won't Have:
- 

### Drawbacks

no known drawbacks

### Alternatives

Our other alternative is to continue using Dropbox, as we deal with so many folders and files, the current use case in Directus won't work for us.

### Adoption Strategy

I think this will improve the user experience in files for everyone

**Comment by w0kyj** [comment]
@divinepersuasionstudio Some additional questions and clarifications:

> Clickable file folder icon inside of file directory with folders displayed first and files displayed second.
- Can this be configurable - eg able to be turned on or off in the Layout settings?
- Do you want or expect the folders to support tree hierarchy views like this:
![Screenshot 2023-05-22 at 3 49 35 PM](https://github.com/directus/directus/assets/40695629/abebfcf8-43b7-4fae-8255-2b22220a9cb4)
    - If yes, do you want/need to reorder file(s) via drag/drop between parent/child folders?
        - eg move file from Grandchild folder to Root folder, etc
        - Is multiple file select and move necessary?


> * custom sorting options
- Does this mean something like multi-column sort?
- Do you mean sort will just reverse the folder/file ascending/descending?
- Something else?


> * Clickability
- If the hierarchy desired above, then:
    - When clicking a child folder/file, should it open the parent or the child?
- Confirming that when clicking a folder, show the same view with this folders contents


Finally - do you want this view style be applied to both the Table and Card layouts?

**Reply by divinepersuasionstudio** [reply]
Hi, 

Yes, I think we would want it to be configurable and support tree hierarchy views. Ideally, I would want to be able to drag and drop between parent and child folders. 

For custom sorting, I was thinking of being able to sort alphabetically (ascending/descending). 

When clicking on a child folder, I would want it to open that folder to view its contents. So the parent folder would show the child folders as well as any individual files in the parent folder. 

I don't think the view style would have to be applied and table and card layouts for us to use it but I like the idea of that being possible. 

Please let me know if you need any additional information!

**Reply by benhaynes** [reply]
I would imagine this is not a "folder" capability, but rather a way to show M2O relationships as GUI "groups". That way this would work for all sorts of data. I thought we had a discussion for this, but I can't find it. I do have some mocks for the design already, though.

The main question is do we build this into each Layout individually, or is there a general capability we can add globally that sits above all Layouts?

**Reply by lukas-schaetzle** [reply]
I think you mean this discussion: [#3464](https://github.com/directus/directus/discussions/4999#discussioncomment-601368)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
